arbnode/mel/extraction/batch_lookup.go

```diff
@@ -12,90 +12,67 @@ import (
 	"github.com/offchainlabs/nitro/solgen/go/bridgegen"
 )
 
-type eventUnpacker interface {
-	unpackLogTo(event any, abi *abi.ABI, eventName string, log types.Log) error
+type EventUnpacker interface {
+	UnpackLogTo(event any, abi *abi.ABI, eventName string, log types.Log) error
 }
 
-func parseBatchesFromBlock(
+func ParseBatchesFromBlock(
 	ctx context.Context,
-	melState *mel.State,
 	parentChainHeader *types.Header,
-	txsFetcher TransactionsFetcher,
-	receiptFetcher ReceiptFetcher,
-	eventUnpacker eventUnpacker,
-) ([]*mel.SequencerInboxBatch, []*types.Transaction, []uint, error) {
-	allBatches := make([]*mel.SequencerInboxBatch, 0)
-	allBatchTxs := make([]*types.Transaction, 0)
-	allBatchTxIndices := make([]uint, 0)
-	parentChainBlockTxs, err := txsFetcher.TransactionsByHeader(ctx, parentChainHeader.Hash())
+	txFetcher TransactionFetcher,
+	logsFetcher LogsFetcher,
+	eventUnpacker EventUnpacker,
+) ([]*mel.SequencerInboxBatch, []*types.Transaction, error) {
+	batches := make([]*mel.SequencerInboxBatch, 0)
+	batchTxs := make([]*types.Transaction, 0)
+	logs, err := logsFetcher.LogsForBlockHash(ctx, parentChainHeader.Hash())
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to fetch transactions for parent chain block %v: %w", parentChainHeader.Hash(), err)
+		return nil, nil, fmt.Errorf("failed to fetch logs from parent chain block %v: %w", parentChainHeader.Hash(), err)
 	}
-	for i, tx := range parentChainBlockTxs {
-		// TODO: remove this temporary work around for handling init message, i.e skipping the check when msgCount==0
-		if melState.MsgCount != 0 {
-			if tx.To() == nil {
-				continue
-			}
-			if *tx.To() != melState.BatchPostingTargetAddress {
```

Comment:
Don't we need the To address filter still? Maybe it's elsewhere and I'm missing it.